src/smt.rs

```diff
@@ -0,0 +1,95 @@
+use std::fmt::{Display, Formatter, Result};
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct SMTVariable {
+    pub name: String,
+}
+
+pub trait SMTSyntaxUnit {
+    fn as_smt(&self) -> String;
+}
+
+impl SMTSyntaxUnit for String {
+    fn as_smt(&self) -> String {
+        self.to_owned()
+    }
+}
+
+pub enum SMTLiteral {
+    Hex(String),
+}
+
+impl SMTLiteral {
+    pub fn hex_zero() -> Self {
+        Self::Hex("0".repeat(64))
+    }
+}
+
+pub enum SMTIdentifier {
+    Simple(String),
+    Indexed(String, usize), // for now supports only one index
+}
+
+impl SMTIdentifier {
+    pub fn bitvec(index: usize) -> Self {
+        Self::Indexed("BitVec".to_owned(), index)
+    }
+}
+
+pub enum SMTExpression {
+    Eq(String, String),
+    Not(String),
+    Assert(String),
```

Comment:
Strictly, Assert, DeclareConst and DefineConst are not expressions but statements. Not sure if this distinction is too important, though.